data/rulesets.ts

```diff
@@ -826,6 +826,15 @@ export const Formats: {[k: string]: FormatData} = {
 			this.add('rule', 'Switch Priority Clause Mod: Faster Pokémon switch first');
 		},
 	},
+	desyncclausemod: {
+		effectType: 'Rule',
+		name: 'Desync Clause Mod',
+		desc: 'In the event a move would cause a desync, the move will fail instead. This rule covers Psywave and Counter.',
+		// The effect itself is implemented in gen1/moves.ts.
+		onBegin() {
+			this.add('rule', 'Desync Clause Mod: Desyncs changed to move failure.');
+		},
```

Comment:
```suggestion
		onBegin() {
			this.add('rule', 'Desync Clause Mod: Desyncs changed to move failure.');
		},
		// Hardcoded in gen1/moves.ts
		// Can't be disabled (no precedent for how else to handle desyncs)
```
(Design principle 1 again; also good to match how other rules are commented.)